ecosystem/purescript-0.14-migration-guide.md

```diff
@@ -0,0 +1,622 @@
+```
+
+Several modules have moved or been renamed, all from the `profunctors` and `bifunctors` packages into the `functors` package. You'll need to update your imports accordingly:
+
+```
+Data.Profunctor.Clown  -> Data.Functor.Clown
+Data.Profunctor.Cowrap -> Data.Functor.Flip
+Data.Profunctor.Joker  -> Data.Functor.Joker
+Data.Profunctor.Costar -> Data.Functor.Costar
+
+Data.Bifunctor.Clown   -> Data.Functor.Clown
+Data.Bifunctor.Flip    -> Data.Functor.Flip
+Data.Bifunctor.Joker   -> Data.Functor.Joker
+Data.Bifunctor.Product -> Data.Functor.Product2
+```
+
+The `Data.Profunctor.Wrap` and `Data.Bifunctor.Wrap` types have been deleted, as any profunctor or bifunctor should also have a functor instance.
+
+This was the old dependency graph:
+
+![old-dependency-graph](https://user-images.githubusercontent.com/1570964/106351480-6e55a100-62aa-11eb-8e8a-72cb33e354dc.png)
+
+This is the new dependency graph, where the dotted line represents three packages which may be merged together in the future:
+
+![new-dependency-graph](https://user-images.githubusercontent.com/1570964/106351479-6ac21a00-62aa-11eb-8234-803f7868cc53.png)
+
+### Notable Library Changes
+
+#### The `Newtype` class now has `Coercible` as a superclass and no longer has `wrap` and `unwrap` class members.
+
+Implemented in [purescript-newtype#22](https://github.com/purescript/purescript-newtype/pull/22).
+
+The `Newtype` class exists to support convenient wrapping and unwrapping and to implement the combinators in the `Data.Newtype` module. The `Newtype` class now has `Coercible` as a superclass, which allows most combinators to now be implemented with `coerce` from the new `safe-coerce` library. This makes these combinators zero-cost for a performance gain.
+
+The `Newtype` class no longer has `wrap` and `unwrap` members. Instead, `wrap` and `unwrap` are synonyms for `coerce` and are exported as functions from the module. Most code does not need to change, but if you manually defined a `Newtype` instance in your code it will no longer compile.
+
+In addition, newtypes that do not have their constructors exported can no longer be unwrapped via the `Newtype` class. Most code will be unaffected, as giving a `Newtype` instance to a newtype with hidden constructors was already considered bad practice and usually accidental.
+
+**To fix:**
+
+Most `Newtype` instances are derived and will continue to work. If you manually defined a `Newtype` instance for a type, however, you can simply remove the `wrap` and `unwrap` definitions:
+
+```diff
++ instance newtypeAdditive :: Newtype (Additive a) a
+- instance newtypeAdditive :: Newtype (Additive a) a where
+-   wrap = Additive
+-   unwrap (Additive a) = a
+```
+
+Most newtypes either export their constructors and have a `Newtype` instance or hide their constructors and do not have the instance. However, if you have a newtype in your code which has hidden constructors and a `Newtype` instance then you will need to either:
```

Comment:
Will you _need_ to do either of these things? I thought this was just a warning for now. Or are we assuming that the reader wants to fix all errors and warnings?

Reply:
I expect the reader will want to at least know how to fix warnings, even if they don’t strictly have to (yet).

Reply:
Right - so do you think it might be worth replacing "need" with "may want to" or something like that?

Reply:
Yes, I'm happy to rephrase to be clear these are warnings, not errors.